app/services/ipam/external_ipam.rb

```diff
@@ -0,0 +1,32 @@
+module IPAM
+  class ExternalIpam < Base
+    delegate :external_ipam_proxy, :to => :subnet
+
+    def suggest_ip
+      if self.mac.nil?
+        errors.add(:mac, "You must specify a MAC address before selecting the External IPAM subnet")
+        return nil
+      end
+
+      logger.debug "Obtaining next available IP from IPAM for subnet #{@subnet.network_address}"
+      response = external_ipam_proxy.next_ip(@subnet.network_address, self.mac)
+
+      if response.key?('error')
+        errors.add(:subnet, response['error'])
+        nil
+      else
+        next_ip = response["data"]
+        logger.debug("IPAM returned #{next_ip} as the next available IP in subnet #{@subnet.network_address}")
+        next_ip
+      end
```

Comment:
Nitpick: I still think that the proxy class should be responsible from error parsing, there should be an exception thrown there and catched here. Not that of a big deal if you prefer this.